timeless/views.py

```diff
@@ -27,3 +27,57 @@ def get(self, *args, **kwargs):
             if result is None:
                 abort(404)
         return "", HTTPStatus.OK
+
+
+class GenericView(views.View):
+    """ Docs http://flask.pocoo.org/docs/1.0/views/ """
+    def dispatch_request(self):
+        """Subclasses have to override this method to implement the
+        actual view function code.  This method is called with all
+        the arguments from the URL rule.
+        """
+        raise NotImplementedError()
+
+    def get_template_name(self):
+        """ Setup of template name """
+        raise NotImplementedError()
+
+
+class ListView(GenericView):
+    def get_objects(self):
+        """ Method for fetching list of objects from db"""
+        raise NotImplementedError()
+
+    def get_template_name(self):
+        """ Setup of template name """
+        raise NotImplementedError()
+
+    def render_template(self, context):
+        """ Render template and provide context """
+        return render_template(self.get_template_name(), **context)
+
+    def dispatch_request(self):
+        """ Dispatch method for managing logic """
+        context = {'objects': self.get_objects()}
+        return self.render_template(context)
+
+
+class DetailView(GenericView):
+    def get_object(self):
+        """ Method for fetching object from db """
+        raise NotImplementedError()
+
+    def get_template_name(self):
+        """ Setup of template name """
+        raise NotImplementedError()
+
+    def render_template(self, context):
+        """ Render template and provide context """
+        return render_template(self.get_template_name(), **context)
+
+    def dispatch_request(self):
+        """ Dispatch method for managing logic """
+        if request.POST:
+            pass
+        context = {'objects': self.get_object()}
+        return self.render_template(context)
```

Comment:
@vladarefiev Could you write some unit tests for the implemented methods? Or is it complex at the moment?

Reply:
@vladarefiev I don't think this topic is finished, maybe you have to leave a puzzle for continuing implementation?

Reply:
@vladarefiev don't forget the remove the existing puzzle from the code.